util: add migration stub

tbg opened the pull request:
Pending #16977, this gives a crude way of landing changes that
will need a proper migration.

The "problem" with this change is that this means that tests won't be run with
the new code completely active. We can change that, but it would approximately
double time spent in CI, so that I'm not convinced it's worth it at this point.
Instead, the "new mode" should be tested thoroughly when it can properly be
migrated into.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17068)
<!-- Reviewable:end -->

tbg committed:
util: add migration stub

Pending #16977, this gives a crude way of landing changes that
will need a proper migration.

The "problem" with this change is that this means that tests won't be run with
the new code completely active. We can change that, but it would approximately
double time spent in CI, so that I'm not convinced it's worth it at this point.
Instead, the "new mode" should be tested thoroughly when it can properly be
migrated into.

tschottdorf commented:
From #16993:

> Yes, IsMigrated looks fine in the short term.

-- @petermattis
